harvdev_utils/db_examine_scripts/genotype_report.py

```diff
@@ -0,0 +1,203 @@
+                                      GenotypeSynonym,
+                                      Phendesc, PhenotypeComparison, Phenstatement)
+from sqlalchemy.orm.session import Session
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--Genotype', help=' (Genotype symbol to make report for)', required=True)
+parser.add_argument('-b', '--by', help='lookup by', choices=['name', 'uniquename'], default='name')
+parser.add_argument('-d', '--debug', help='dump out all debug messages', default=False, action='store_true')
+parser.add_argument('-c', '--config', help='Specify the location of the configuration file.', required=True)
+parser.add_argument('-l', '--limit', help='Limit to x for each table', type=int, default=0, required=False)
+args = parser.parse_args()
+
+log = logging.getLogger(__name__)
+
+# Import secure config variables.
+config = configparser.ConfigParser()
+config.read(args.config)
+
+if args.debug:
+    print("Running in Debug mode")
+    logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format='%(levelname)s -- %(message)s')
+    logging.getLogger('sqlalchemy.engine').setLevel(logging.DEBUG)
+    # comment/uncomment out below to notsee/see NOTICE messages for sql functions.
+    # logging.getLogger('sqlalchemy.dialects.postgresql').setLevel(logging.INFO)
+else:
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO, format='%(levelname)s -- %(message)s')
+
+
+def get_Genotype(session: Session, symbol: str, lookup_by: str) -> Genotype:
+    """Lookup Genotype."""
+    log.info("Looking up Genotype '{}' using method '{}'".format(symbol, lookup_by))
+    try:
+        if lookup_by == 'name':
+            log.info("Ignoring lookup by name as only unique names are set.")
+            return session.query(Genotype).filter(Genotype.uniquename == symbol).one()
+        elif lookup_by == 'uniquename':
+            return session.query(Genotype).filter(Genotype.uniquename == symbol).one()
+    except NoResultFound:
+        log.info("Could NOT find '{}' by uniquename. exiting".format(symbol))
+        exit(-1)
+    except MultipleResultsFound:
+        log.info("Could NOT find UNIQUE entry for '{}' by uniquename. exiting".format(symbol))
+        exit(-1)
+
+
+def report_Genotype(session: Session, symbol: str, debug: bool, limit: int, lookup_by: str):  # noqa
+    """Write report."""
+    ###########################
+    # starting point for report
+    ###########################
+    Genotype = get_Genotype(session, symbol, lookup_by)
+    print("BOB: {} {}".format(type(Genotype), Genotype.__class__.__bases__))
+    log.info("###################### Genotype ############################")
+    log.info(Genotype)
+    log.info("###########################################################")
+    # log.info(dir(feature))
+
+    log.info("\n################### GenotypePub #####################")
+    fps = session.query(GenotypePub).filter(GenotypePub.genotype_id == Genotype.genotype_id)
```

Comment:
Should be `gps` not `fps`? This is a confusing var name.

Reply:
Due to a cut and paste from cell_line which in turn was a cut and paste from feature hence the fps.